### ISSUE TITLE
Your application could not be compiled, because its dependencies could not be established.

### ISSUE THREAD

**Issue by ztz12** [flutter/flutter#25739]
when i create a new dart file and run android studio,it throws Your application could not be compiled, because its dependencies could not be established this probleam,i have no idea what can lead to it.

**Comment by GaryQian** [comment]
Are you creating a flutter project with the Android Studio plugin or just a dart file by itself? Flutter should resolve dependencies such as dart if you are making a flutter project.

**Comment by zoechi** [comment]
Please add the output of `flutter doctor -v`.

**Comment by ztz12** [comment]
this is my flutter doctor result
flutter doctor -v
[✓] Flutter (Channel stable, v1.0.0, on Mac OS X 10.14.2 18C54, locale
    zh-Hans-CN)
    • Flutter version 1.0.0 at /Users/zhangtianzhu/flutter
    • Framework revision 5391447fae (4 weeks ago), 2018-11-29 19:41:26 -0800
    • Engine revision 7375a0f414
    • Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at /Users/zhangtianzhu/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling
      support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_152-release-1024-b01)
    • All Android licenses accepted.

[✗] iOS toolchain - develop for iOS devices
    ✗ Xcode installation is incomplete; a full installation is necessary for iOS
      development.
      Download at: https://developer.apple.com/xcode/download/
      Or install Xcode via the App Store.
      Once installed, run:
        sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
    ✗ libimobiledevice and ideviceinstaller are not installed. To install with
      Brew, run:
        brew update
        brew install --HEAD usbmuxd
        brew link usbmuxd
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    ✗ ios-deploy not installed. To install with Brew:
        brew install ios-deploy
    ✗ CocoaPods not installed.
        CocoaPods is used to retrieve the iOS platform side's plugin code that
        responds to your plugin usage on the Dart side.
        Without resolving iOS dependencies with CocoaPods, plugins will not work
        on iOS.
        For more info, see https://flutter.io/platform-plugins
      To install:
        brew install cocoapods
        pod setup

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 28.0.1
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build
      1.8.0_152-release-1024-b01)

[✓] VS Code (version 1.29.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.20.0

[✓] Connected device (1 available)
    • Redmi 4X • 3ba790f77ce4 • android-arm64 • Android 7.1.2 (API 25)

! Doctor found issues in 1 category.

**Comment by ztz12** [comment]
when i run Android studio ，it show this [pic](https://ztz12yif.oss-cn-hongkong.aliyuncs.com/pic.png?Expires=1545965177&OSSAccessKeyId=TMP.AQGx3bE6wl0koa5QvaYdTnYh06yW12gtCCEtHK4SFgw1CNi_6CJ6AxMkqDAmADAtAhUAqXGlkEaTRG1Hlcof9PB-CUCjdBUCFC1WCWNMJpPvzrLtU2qUIw6iCCa3&Signature=YH7WYhhkZejZZrSmdWqxczC%2B6B0%3D)

**Comment by kangwang1988** [comment]
@ztplz 
As GaryQian said, we suggest you to develop flutter in a flutter project.
I suggest you create a new flutter project and write codes in it. As the flutter plugin can manage the dependencies for you with pubspec.yaml and flutter packages get, it might solve this problem. If it remains, paste error message here please.

**Comment by ztplz** [comment]
@kangwang1988 Oops, you ping the wrong person.

**Comment by kangwang1988** [comment]
@ztplz  Sorry.

**Comment by ztz12** [comment]
I create a new flutter project and write code in it ,this problem offen happen
[pic](https://yifimage.oss-cn-shanghai.aliyuncs.com/picture.png?Expires=1545970372&OSSAccessKeyId=TMP.AQGx3bE6wl0koa5QvaYdTnYh06yW12gtCCEtHK4SFgw1CNi_6CJ6AxMkqDAmADAtAhUAqXGlkEaTRG1Hlcof9PB-CUCjdBUCFC1WCWNMJpPvzrLtU2qUIw6iCCa3&Signature=TVoN%2F3fhnxmFTMkUrJwUukD4hFs%3D)

**Comment by kangwang1988** [comment]
Do tests following those steps:
1. Delete the pubspec.lock
So that dart will re-get corresponding dependencies specified in your pubspec.yaml
2. Using `flutter packages get -v` from the terminal to see if it works fine
In fact, the error message resulted from this logic below:
<img width="1228" alt="screen shot 2018-12-28 at 11 33 05 am" src="https://user-images.githubusercontent.com/817851/50501613-61936d00-0a94-11e9-9bf7-b86418fdf18b.png">

In other words, flutter packages get -v failed.
Make sure it works fine from the terminal.

**Comment by zoechi** [comment]
`flutter packages upgrade` should have the same effect as deleting `pubspec.lock` and running `flutter packages get`

@ztz12 

> because its dependencies could not be established

Please add the full and exact error output.

**Comment by zoechi** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. We are therefore reluctantly going to close this bug for now. Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!
Thanks for your contribution.

**Comment by wbdsnx** [comment]
I am facing with the same issue,have you solved this? If yes,please tell me how I can do .
Thank you !

**Comment by zoechi** [comment]
@wbdsnx 
Please consider asking support questions in one of the other channels listed at http://flutter.io/support .

**Comment by ztz12** [comment]
l hava no idea to solve it, and I suspect it is my Apple system problem.

**Comment by wbdsnx** [comment]
what can you do when you need to create a dart file ,facing with such a issue like this above ?

**Comment by wbdsnx** [comment]
@ztz12 what can you do when you need to create a dart file ,facing with such a issue like this above ?

**Comment by ztz12** [comment]
if i create a new dart ,it show this problem , so i cannot create a new dart . i use my windows computer to make flutter project

**Comment by zoechi** [comment]
Where does this message come from? Is this from the network proxy?
Flutter needs network access to create a new project, especially when run from within AS.
Please try if `flutter create foo --offline` works better.

**Comment by wbdsnx** [comment]
so when you create a new dart file ,you always put it into another flutter project to get dependency?

**Comment by bugudiramu** [comment]
Your application could not be compiled, because its dependencies could not be established.
When trying to parse this Dart file to find its dependencies:
  F:\flutter\.pub-cache\hosted\pub.dartlang.org\http-0.12.0+1\lib\http.dart
...the analyzer failed with the following error:
  Error in F:\flutter\.pub-cache\hosted\pub.dartlang.org\http-0.12.0+1\lib\http.dart: Expected to find ';'.
Exited (sigterm)

**Comment by zoechi** [comment]
@bugudiramu try `flutter packages pub cache repair`

**Comment by bugudiramu** [comment]
It just worked.Thank you so much it took me 2 days for figuring out the
solution.Thank you.

**Comment by junqit** [comment]
更新自己的flutter 稳定版的库

**Comment by alexwan02** [comment]
> 2\. flutter packages get -v

it works

**Comment by alexwan02** [comment]
> Do tests following those steps:
> 
> 1. Delete the pubspec.lock
>    So that dart will re-get corresponding dependencies specified in your pubspec.yaml
> 2. Using `flutter packages get -v` from the terminal to see if it works fine
>    In fact, the error message resulted from this logic below:
> 
> <img alt="screen shot 2018-12-28 at 11 33 05 am" width="1228" src="https://user-images.githubusercontent.com/817851/50501613-61936d00-0a94-11e9-9bf7-b86418fdf18b.png">
> 
> In other words, flutter packages get -v failed.
> Make sure it works fine from the terminal.

It works

**Comment by jonahwilliams** [comment]
This is generally an issue with typos in import statements or missing pubspec dependencies. It's not really clear that there are actionable items from this issue, so I'm closing it. If you are following this and have a similar issue, please file a new bug with the requisite information filled in.
